### PR TITLE
Added Map File to file structure.

### PR DESCRIPTION
**Pull request by thehenrytsai**
1. Added Map File to file structure.
1. Increased code coverage of `TransactionProcessor` to 100%.
1. Some code refactoring.

### CODE REVIEW

File: tests/core/AnchorFile.spec.ts [hunk 1]
```diff
@@ -12,140 +13,146 @@ describe('AnchorFile', async () => {
...
 
-      await expectAsync(AnchorFile.parseAndValidate(anchorFileCompressed, 1))
+      // Overreid the default max operation count per batch.
+      const originalMaxOperationsPerBatch = ProtocolParameters.maxOperationsPerBatch;
+      ProtocolParameters.maxOperationsPerBatch = 1;
+
+      await expectAsync(AnchorFile.parseAndValidate(anchorFileCompressed))
         .toBeRejectedWith(new SidetreeError(ErrorCode.AnchorFileExceededMaxOperationCount));
+
+      ProtocolParameters.maxOperationsPerBatch = originalMaxOperationsPerBatch;
```
Comment (mudiali) [inline tests/core/AnchorFile.spec.ts]: Sanity check: should this be in 'try-finally'? Not sure if the expectAsync above fails then whether this line is executed or not.

File: tests/core/MapFile.spec.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import Compressor from '../../lib/core/versions/latest/util/Compressor';
+import ErrorCode from '../../lib/core/versions/latest/ErrorCode';
+import JasmineSidetreeErrorValidator from '../JasmineSidetreeErrorValidator';
+import MapFile from '../../lib/core/versions/latest/MapFile';
+import MapFileModel from '../../lib/core/versions/latest/models/MapFileModel';
+import SidetreeError from '../../lib/core/SidetreeError';
+
+describe('MapFile', async () => {
```
Comment (mudiali) [inline tests/core/MapFile.spec.ts]: Sanity: there's no positive test case ... shouldn't we add one?

File: tests/core/AnchorFile.spec.ts (hunk 1 above)
Comment (thehenrytsai) [reply tests/core/AnchorFile.spec.ts]: Thanks, good call!

File: tests/core/MapFile.spec.ts (hunk 2 above)
Comment (thehenrytsai) [reply tests/core/MapFile.spec.ts]: Discuss offline, `Observer` "E2E" test already covers this path, so will skip adding for this case.

**Pull request merged**